src/gun.erl

```diff
@@ -1780,8 +1780,12 @@ disconnect_flush(State=#state{socket=Socket, messages={OK, Closed, Error}}) ->
 active(State=#state{active=false}) ->
 	State;
 active(State=#state{socket=Socket, transport=Transport}) ->
-	Transport:setopts(Socket, [{active, once}]),
-	State.
+	case Transport:setopts(Socket, [{active, once}]) of
+		{error, closed} ->
```

Comment:
What about other error conditions?

Reply:
I didn't solve them in this PR. :-) I assume they can be handled in the same way, but I don't have any particular case right now.

Reply:
Note that `closed` is not a network error. It's just that the server has closed the socket and we may have missed it while we were in passive mode.

Reply:
Sure but you can get a socket closed for other reasons so if we're going to handle one case we should handle all of them.

Reply:
If it's as easy as adding another case `{error, _} = Error -> disconnect(State, Error)`, then sure. I checked now and it doesn't break any test cases at least. Would this be enough?

If this gets bigger, I think having no errors in the Cowboy test suites is quite nice to start with, so you can rely on the build results for PRs...

Reply:
Done.